File: MoonApp/packages/react-native-nitro-moon/src/module/MRTC.ts
import { NitroModules } from 'react-native-nitro-modules'
import type { MRTC as MRTCType  } from '../specs/MRTC/MRTC.nitro'

export type { MRTCLiveStream, MRTCIceServer } from '../specs/MRTC/MRTC.nitro'
export * from '../specs/MRTC/MRTCPeer.nitro'
export const MRTC = NitroModules.createHybridObject<MRTCType>('MRTC')

File: MoonApp/packages/react-native-nitro-moon/ios/MRTC/MRTC.swift
import Foundation
import NitroModules
import WebRTC


final class MRTC : HybridMRTCSpec {
    
    // 直播流画面
    static var liveStream: MRTCLiveStream = MRTCLiveStream(
        width: 1280,
        height: 720,
        framerate: 15,
        bitrate: 800_00,
        scalabilityMode: nil,
        bitratePriority: nil,
        networkPriority: nil,
        preference: nil
    )
    
    // TURN 服务器
    static var iceServers: [ RTCIceServer ] = []
    
    // 设置直播流画面
    func setLiveStream(stream: MRTCLiveStream) {
        print(stream)
        MRTC.liveStream = stream
    }
    
    // 设置 TURN 服务器
    func setIceServers(servers: [ MRTCIceServer ]) {
        MRTC.iceServers = servers.map { $0.rtcValue }
    }
    
    // 创建 WebRTC 连接
    func createPeer() -> any HybridMRTCPeerSpec {
        return MRTCPeer()
    }
}
